src/Common/src/CoreLib/System/IO/PathInternal.cs

```diff
@@ -117,24 +117,54 @@ internal static bool AreRootsEqual(string first, string second, StringComparison
         internal static string RemoveRelativeSegments(string path, int rootLength)
         {
             Debug.Assert(rootLength > 0);
-            bool flippedSeparator = false;
+            
+            Span<char> initialBuffer = stackalloc char[260 /* PathInternal.MaxShortPath */];
```

Comment:
Why use the explicit constant value rather than using MaxShortPath?  Is that because this is in shared code and MaxShortPath is only defined for Windows?  Ideally there'd be a better way to address that than duplicating the const value.

Reply:
It would probably be best to create a new constant like `PathInternal.MaxStackPath` or something like that. You can comment on that one saying something like:

``` c#
<summary>
 Amount of stack space to use when handling paths.
</summary>
<remarks>
 We're using 260 here as it is both a reasonable stack size and is the historical MAX_PATH for Windows. Most paths should fit.
</remarks>
```